### ISSUE TITLE
SHARED_LINKER_FLAGS / EXE_LINKER_FLAGS equivalent

### ISSUE THREAD

**Issue by VA-GS** [mesonbuild/meson#13790]
Hello, I am not familiar with Meson but it is use by one of my dependency that I consume using Conan. In that context, I found out that the sharedlinkflags/exelinkflags that Conan exposes to consumer to pass additional linker flags to dependencies is not respected when the dependency builds with Meson. The reason is that the code in Conan concatenates both into cpp_link_args (see https://github.com/conan-io/conan/issues/17161).

Is there a way to pass those to Meson in a way more similar to CMAKE SHARED_LINKER_FLAGS / EXE_LINKER_FLAGS ?

**Comment by dcbaker** [comment]
No, we don't currently have anything like that. It probably wouldn't be out of scope to have `<lang>_link_args_<shared|static|exe>`, especially since we have the ability to differentiate the shared and static at the DSL level, just not in a way exposed vial machine files or command line.
